airbyte-scheduler/client/src/main/java/io/airbyte/scheduler/client/DefaultSynchronousSchedulerClient.java

```diff
@@ -55,6 +63,7 @@ public SynchronousResponse<StandardCheckConnectionOutput> createSourceCheckConne
 
     return execute(
         ConfigType.CHECK_CONNECTION_SOURCE,
+        jobCheckConnectionConfig,
         source.getSourceDefinitionId(),
         jobId -> temporalClient.submitCheckConnection(UUID.randomUUID(), 0, jobCheckConnectionConfig),
```

Comment:
Yes it makes senses:
- Workflow ID are unique in temporal and multiple check connection should be able to run in parallel for the same workspace and even more for the same actor definition.
- When running the check connection, the actor doesn't exist nor the connection so we don't have a id discriminant enough to use instead of a random UUID.
- For the sync and connection manager workflow, we want unicity and we have discriminant ids,

Reply:
I noticed we were generating two UUIDs and using one for segment tracking and generating another for the actual job (even though the UUID generated for segment was being passed), so I updated this logic a bit as part of [833ddcb](https://github.com/airbytehq/airbyte/pull/14818/commits/833ddcb7eaa83ac715b2f109261875fbf7e99d1a) to make sure the same uuid is being used

Reply:
Sounds good, thanks!